operator/operator.go

```diff
@@ -159,25 +144,94 @@ func NewOperatorFromConfig(c optypes.NodeConfig) (*Operator, error) {
 		panic(err)
 	}
 
-	txSender, err := wallet.NewPrivateKeyWallet(ethRpcClient, signerV2, common.HexToAddress(c.OperatorAddress), logger)
+	txSender, err := wallet.NewPrivateKeyWallet(ethHttpClient, signerV2, common.HexToAddress(c.OperatorAddress), logger)
 	if err != nil {
 		logger.Error("Failed to create transaction sender", "err", err)
 		return nil, err
 	}
 
-	txMgr := txmgr.NewSimpleTxManager(txSender, ethRpcClient, logger, common.HexToAddress(c.OperatorAddress)).WithGasLimitMultiplier(1.5)
-
-	// Use eigen registry
-	reg = sdkClients.PrometheusRegistry
+	txMgr := txmgr.NewSimpleTxManager(txSender, ethHttpClient, logger, common.HexToAddress(c.OperatorAddress)).WithGasLimitMultiplier(1.5)
 
 	registryCoordinatorAddress := common.HexToAddress(c.AVSRegistryCoordinatorAddress)
-	aggregatorRpcClient, err := NewAggregatorRpcClient(c.AggregatorServerIpPortAddress, operatorId, registryCoordinatorAddress, logger)
+	operatorStateRetrieverAddress := common.HexToAddress(c.OperatorStateRetrieverAddress)
+
+	avsRegistryContractBindings, err := chainioutils.NewAVSRegistryContractBindings(
+		registryCoordinatorAddress,
+		operatorStateRetrieverAddress,
+		ethHttpClient,
+		logger,
+	)
+	if err != nil {
+		logger.Error("Cannot create AVSRegistryContractBindings", "err", err)
+		return nil, err
+	}
+
+	avsRegistryChainReader := chainioavsregistry.NewAvsRegistryChainReader(
+		avsRegistryContractBindings.RegistryCoordinatorAddr,
+		avsRegistryContractBindings.BlsApkRegistryAddr,
+		avsRegistryContractBindings.RegistryCoordinator,
+		avsRegistryContractBindings.OperatorStateRetriever,
+		avsRegistryContractBindings.StakeRegistry,
+		logger,
+		ethHttpClient,
+	)
+
+	delegationManagerAddr, err := avsRegistryContractBindings.StakeRegistry.Delegation(&bind.CallOpts{})
+	if err != nil {
+		logger.Fatal("Failed to fetch Slasher contract", "err", err)
+		return nil, err
+	}
+
+	avsDirectoryAddr, err := avsRegistryContractBindings.ServiceManager.AvsDirectory(&bind.CallOpts{})
+	if err != nil {
+		logger.Fatal("Failed to fetch Slasher contract", "err", err)
+		return nil, err
+	}
+
+	elContractBindings, err := chainioutils.NewEigenlayerContractBindings(
+		delegationManagerAddr,
+		avsDirectoryAddr,
+		ethHttpClient,
+		logger,
+	)
+	if err != nil {
+		logger.Fatal("Failed to create EigenlayerContractBindings", "err", err)
+		return nil, err
+	}
+
+	elChainReader := elcontracts.NewELChainReader(
+		elContractBindings.Slasher,
+		elContractBindings.DelegationManager,
+		elContractBindings.StrategyManager,
+		elContractBindings.AvsDirectory,
+		logger,
+		ethHttpClient,
+	)
+
+	elChainWriter := elcontracts.NewELChainWriter(
+		elContractBindings.Slasher,
+		elContractBindings.DelegationManager,
+		elContractBindings.StrategyManager,
+		elContractBindings.StrategyManagerAddr,
+		elChainReader,
+		ethHttpClient,
+		logger,
+		nil,
+		txMgr,
+	)
```

Comment:
Not sure how I feel about this - I get the intention of not constructing things that are not needed, but I feel like this may be getting a bit much. Maybe a method such as `createOperatorClients` or such would be interesting here.
Either way, not blocking.

Reply:
I ended up encapsulating some of these constructors into more minimal APIs. The code now is much shorter and implementation details are hidden.